### PR TITLE
Add `blockchain_wallet.get` and `blockchain_wallet.all`

### LINKED ISSUE #691: Management endpoints for blockchain wallets
Depends on #690 

In order to manage the blockchain wallets, the following endpoints need to be added to the Admin API.

- `/blockchain_wallet.get`
- `/blockchain_wallet.all`

### PR DESCRIPTION
**Pull request by ripzery**
Issue/Task Number: #691 

Closes #691 

# Overview

This PR adds 2 endpoints for blockchain wallet which are:
- `blockchain_wallet.get` to  get a single wallet by the given address
- `blockchain_wallet.all` to get all wallets by given pagination params

# Usage

Follows [Setup local geth testnet guide](https://github.com/omisego/ewallet/tree/master/tools/local_geth_testnet) then you will need to create a wallet and update tokens' `blockchain_address`:

1. `mix seed --sample`
2. `iex -S mix`
3. Insert blockchain wallet by run the followings command:
```elixir
EWalletDB.BlockchainWallet.insert(
  %{
    address: "0xprimary_address", 
    type: "hot", 
    name: "Primary wallet", 
    public_key: "12345", 
    originator: %ActivityLogger.System{}
  }
)
```
Replace address with your wallet address from the geth setup.

4. Update Ethereum's `blockchain_address`
```elixir
EWalletDB.Token.update(
  EWalletDB.Token.get_by(name: "Ether"), 
  %{
    blockchain_address: "0x0000000000000000000000000000000000000000", 
    originator: %ActivityLogger.System{}
  }
)
```
 5. Update OmiseGO's `blockchain_address`
```elixir
EWalletDB.Token.update(
  EWalletDB.Token.get_by(name: "OmiseGO"), 
  %{
    blockchain_address: "omg_address", 
    originator: %ActivityLogger.System{}
  }
)
```
Replace `omg_address` with the omg token address from geth (Run `omgtoken.address`)

6. Try retrieve a wallet or a list of wallets from `/blockchain_wallet.get` and `/blockchain_wallet.all` in the [Swagger](http://localhost:4000/api/admin/docs.ui#/)

### CODE REVIEW

File: apps/ewallet/lib/ewallet/web/v1/overlays/blockchain_wallet_overlay.ex [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      address: nil,
+      name: nil,
+      inserted_at: nil,
+      created_at: nil
+    ]
+
+  def pagination_fields,
+    do: [
+      :address,
+      :inserted_at,
```
Comment (unnawut) [inline apps/ewallet/lib/ewallet/web/v1/overlays/blockchain_wallet_overlay.ex]: Also name?

File: apps/ewallet/test/ewallet/fetchers/blockchain_balance_fetcher_test.exs [hunk 2]
```diff
@@ -40,6 +40,54 @@ defmodule EWallet.BlockchainBalanceFetcherTest do
...
+      token_1 =
+        insert(:token, %{blockchain_address: "0x0000000000000000000000000000000000000000"})
+
+      token_2 =
+        insert(:token, %{blockchain_address: "0x0000000000000000000000000000000000000001"})
+
+      wallet_addresses = [blockchain_wallet_1.address, "0x0"]
+      tokens = [token_1, token_2]
+
+      assert {:error, :blockchain_adapter_error, [error: ":invalid_address"]} =
```
Comment (unnawut) [inline apps/ewallet/test/ewallet/fetchers/blockchain_balance_fetcher_test.exs]: The colon'ed atom is Elixir specific. Can we clean it up?

File: apps/ewallet/test/ewallet/web/blockchain_balance_loader_test.exs [hunk 3]
```diff
@@ -47,10 +47,64 @@ defmodule EWallet.Web.BlockchainBalanceLoaderTest do
...
+      assert {:ok, blockchain_wallet_with_balances} =
+               BlockchainBalanceLoader.wallet_balances(blockchain_wallet, [token_1, token_2])
+
+      assert blockchain_wallet_with_balances.address == blockchain_wallet.address
+      assert [balance_token_1, balance_token_2] = blockchain_wallet_with_balances.balances
+      assert balance_token_1 == %{token: token_1, amount: 123}
+      assert balance_token_2 == %{token: token_2, amount: 123}
+    end
+
+    test "returns multiple wallets wil balances when given multiple wallets and tokens" do
```
Comment (unnawut) [inline apps/ewallet/test/ewallet/web/blockchain_balance_loader_test.exs]: wil

**Comment by unnawut** [comment]
Should `BlockchainWallet` now enforce association with `Account` or `User`? Right now it’s floating without any association so I’m not sure what should be returned/unauthorized for the endpoints.

Also, `EWalletDB.GlobalRole` and `EWalletDB.Role` are empty at the moment. Probably needed to make these 2 endpoints fully functioning.

File: apps/ewallet_db/test/ewallet_db/blockchain_wallet_test.exs [hunk 4]
```diff
@@ -29,29 +29,48 @@ defmodule EWalletDB.BlockchainWalletTest do
...
+    test "allows insert if provided a user without account_uuid" do
+      {res, _wallet} =
+        :blockchain_wallet
+        |> params_for(%{user: insert(:user), account_uuid: nil})
+        |> BlockchainWallet.insert()
+
+      assert res == :ok
+    end
+
+    test "allows insert if provided an account without user" do
```
Comment (unnawut) [inline apps/ewallet_db/test/ewallet_db/blockchain_wallet_test.exs]: Also needs a failed case where both user and account are given

**Comment by unnawut** [comment]
Also this probably needs an update: `EWallet.BlockchainWalletPolicy` (see `EWallet.WalletPolicy`)

**Comment by ripzery** [comment]
> Also this probably needs an update: `EWallet.BlockchainWalletPolicy` (see `EWallet.WalletPolicy`)

I think it needs to be updated when the `blockchain_wallet.create` is available

File: apps/admin_api/lib/admin_api/v1/controllers/blockchain_wallet_controller.ex [hunk 5]
```diff
@@ -91,23 +84,11 @@ defmodule AdminAPI.V1.BlockchainWalletController do
...
-      _ -> paginator_wallets
-    end
-  end
-
   defp respond_single(conn, blockchain_wallet) do
     render(conn, :blockchain_wallet, %{blockchain_wallet: blockchain_wallet})
   end
 
-  defp respond_multiple(conn, %Paginator{} = paged_wallets) do
+  defp respond_multiple(%Paginator{} = paged_wallets, conn) do
```
Comment (unnawut) [inline apps/admin_api/lib/admin_api/v1/controllers/blockchain_wallet_controller.ex]: Do you mind switching the params for respond_single above too? 🙏 Not very useful but better for consistency.

File: apps/admin_api/priv/spec.json [hunk 6]
```diff
@@ -32523,20 +32523,10 @@
                       "example": {
                         "data": {
                           "object": "blockchain_wallet",
-                          "address": 4.845583031631242e+47,
+                          "address": "0x54e0588607dcec6c0b36fca1154a57814a913591",
```
Comment (unnawut) [inline apps/admin_api/priv/spec.json]: 😂

**Pull request merged**